### ISSUE TITLE
in screencast mode filter arrow key press

### ISSUE THREAD

**Issue by sanket-bhalerao** [microsoft/vscode#103310]
Issue Type: <b>Bug</b>

in screencast mode pressing arrow key shows up on the overlay.
can you add some setting to avoid this?
i am fine with it shown as "ctrl + upArrow"
currently its shown as "upArrow upArrow downArrow upArrow upArrow downArrow". 

VS Code version: Code - Insiders 1.48.0-insider (731f9c25632dbbf01ee3a7892ad9d2791fe0260c, 2020-07-24T05:57:10.466Z)
OS version: Windows_NT x64 10.0.18363

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-8265U CPU @ 1.60GHz (8 x 1800)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.87GB (9.10GB free)|
|Process Argv|. --crash-reporter-id 340e0eb9-5b34-4206-9e2e-bd86fc3b2cf3|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (29)</summary>

Extension|Author (truncated)|Version
---|---|---
ng-template|Ang|0.1000.4
vscode-eslint|dba|2.1.8
gitlens|eam|10.2.2
EditorConfig|Edi|0.15.1
vscode-npm-script|eg2|0.3.12
prettier-vscode|esb|5.1.3
go-nightly|gol|2020.7.2222
output-colorizer|IBM|0.1.2
mongodb-vscode|mon|0.1.0
gather|ms-|2020.7.0
python|ms-|2020.7.96456
vscode-pylance|ms-|2020.7.3
remote-containers|ms-|0.128.0
remote-ssh|ms-|0.51.0
remote-ssh-edit|ms-|0.51.0
remote-wsl|ms-|0.44.4
vscode-remote-extensionpack|ms-|0.20.0
typescript-javascript-grammar|ms-|0.0.53
vscode-typescript-tslint-plugin|ms-|1.2.3
cordova-tools|msj|1.9.4
debugger-for-chrome|msj|4.12.9
fabric8-analytics|red|0.1.0
vscode-xml|red|0.13.0
vscode-yaml|red|0.9.1
sonarlint-vscode|Son|1.16.0
code-spell-checker|str|1.9.0
vscode-stylelint|sty|0.84.0
vscodeintellicode|Vis|1.2.9
quokka-vscode|Wal|1.0.311

</details>

**Comment by sanket-bhalerao** [comment]
@joaomoreno vscode already have filter for single key press, will this not come under that setting itself?

**Comment by joaomoreno** [comment]
Do you want to hide arrow keys? Do you want to not show repetitions? I don't really understand what the issue is.

**Comment by sanket-bhalerao** [comment]
hide the arrow keys when the only purpose is to traverse code.
ex: if user is using up arrow key to go up a line -> hide it.
if user is using ctrl+rightArrow to go to end of current word  - > show it.

**Comment by sanket-bhalerao** [comment]
@joaomoreno can you confirm if you understood the issue?

**Comment by joaomoreno** [comment]
> hide the arrow keys when the only purpose is to traverse code.

But this is sometimes the exact reason why some people would like to show the keybindings, so the viewer can see which arrow key was pressed, as code is navigated around.

I'll keep this closed as designed, sorry!

**Comment by charles-allen** [comment]
As a university lecturer, this feature "as designed" is completely unusable. It feels like this was never tested by a real-world user.

I wanted to show shortcuts like Ctrl+S. Instead my screen is completely spammed.  Just look at this... All I'm doing is moving the caret. How do you imagine this being useful?

![Screen Shot 2021-07-08 at 21 06 48](https://user-images.githubusercontent.com/31621095/124936256-92bf6180-e030-11eb-86af-f8ed7c6f0f6f.png)

**Comment by charles-allen** [comment]
@joaomoreno - As a university lecturer, this feature "as designed" isn't worth using. It feels like this wasn't tested in the wild.

I want to show shortcuts like Ctrl+S. Instead my screen is completely spammed.  Just look at this... All I'm doing is moving the caret. Who finds this useful?

<img src="https://user-images.githubusercontent.com/31621095/124936256-92bf6180-e030-11eb-86af-f8ed7c6f0f6f.png" width="300" />

Edit: It seems that screenshot is an unfair representation; it's because I keep changing direction (which after reviewing my class videos is not realistic). Here are some real-world cases taken straight from my classes:

**TL;DR: Filtering arrows & shift+arrows would probably solve 99% of my issues.**

<img src="https://user-images.githubusercontent.com/31621095/124939207-1f6b1f00-e033-11eb-9b79-fd3a6d0e2dfa.gif" width="600" />

<img src="https://user-images.githubusercontent.com/31621095/124939226-242fd300-e033-11eb-8a00-03877f951c7e.gif" width="600" />

<img src="https://user-images.githubusercontent.com/31621095/124939235-25610000-e033-11eb-8ad9-73d4a55401ec.gif" width="600" />

**Comment by charles-allen** [comment]
Just to add... I also think the expectation for a setting called "Screencast Mode: Only Keyboard Shortcuts" is to filter arrows and backspace; they are literally single key presses performing their global default purpose. IMO that is not the definition of a shortcut.

**Comment by gitAcceptance** [comment]
How could this ever be helpful to my fellow developers?
![image](https://user-images.githubusercontent.com/8495299/185638284-ddefecc4-1368-4470-8386-60dc754f5751.png)

**Comment by webarthur** [comment]
I thing this bug is located at: /src/vs/workbench/browser/actions/developerActions.ts

Maybe in the line 227 `disposables.add(onKeyDown.event(e => {`...

When I have more time I will continue to investigate...

**Comment by webarthur** [comment]
Hy guys! I have added the lines to fix this issue and sent a pull request:

![image](https://user-images.githubusercontent.com/100484/213598442-9fd38785-76d1-4cbf-9569-948ba711f79f.png)

Cheers!

**Comment by webarthur** [comment]
I have added some features to screencast mode:

* Fix arrow key repetition
* Fix label of pressed arrow keys
* Add option to hide single arrow keys pressed

Sent a new pull request too:

https://github.com/microsoft/vscode/pull/171845
